delta-lake/src/test/java/org/apache/iceberg/delta/TestDeltaLakeTypeToType.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.delta;
+
+import io.delta.standalone.types.ArrayType;
+import io.delta.standalone.types.BinaryType;
+import io.delta.standalone.types.BooleanType;
+import io.delta.standalone.types.DoubleType;
+import io.delta.standalone.types.LongType;
+import io.delta.standalone.types.MapType;
+import io.delta.standalone.types.NullType;
+import io.delta.standalone.types.StringType;
+import io.delta.standalone.types.StructType;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.exceptions.ValidationException;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.assertj.core.api.Assertions;
+import org.junit.Before;
+import org.junit.Test;
+
+public class TestDeltaLakeTypeToType {
+  private static final String optionalBooleanType = "testNullableBoolType";
+  private static final String requiredBinaryType = "testRequiredBinaryType";
+  private static final String doubleArrayType = "testNullableArrayType";
+  private static final String innerAtomicSchema = "testInnerAtomicSchema";
+  private static final String stringLongMapType = "testStringLongMap";
+  private static final String nullType = "testNullType";
+  private StructType deltaAtomicSchema;
+  private StructType deltaNestedSchema;
+  private StructType deltaShallowNullTypeSchema;
+  private StructType deltaNullTypeSchema;
+
+  @Before
+  public void constructDeltaLakeSchema() {
+    deltaAtomicSchema =
+        new StructType()
+            .add(optionalBooleanType, new BooleanType())
+            .add(requiredBinaryType, new BinaryType(), false);
+    deltaNestedSchema =
+        new StructType()
+            .add(innerAtomicSchema, deltaAtomicSchema)
+            .add(doubleArrayType, new ArrayType(new DoubleType(), true), false)
```

Comment:
wondering if we want to have test coverage around array of struct type

Reply:
Thank you for your suggestion. I added a `structArrayType` here and also an array of struct column in the nestedDataFrame used in the integration test.